added careful copy job with corresponding test.

ManInFez opened the pull request:
**Task**
#416 


**Approach**
_Short description of the approach_


**Pre un-WIP checklist**
- [x] Statoil tests pass locally
- [ ] Have completed graphical integration test steps

**Depends on**
* Statoil/libecl#

Commit:
added careful copy job with corresponding test.

joakim-hove commented on python/res/fm/shell/shell.py:
```diff
@@ -212,3 +212,9 @@ def copyFile(src , target = None):
+    @staticmethod
+    def carefulCopyFile(src , target = None):
+        if os.path.exists(target):
+            raise IOError("Input argument:'%s' already exist" % target)
```
That the target file exists is perfectly OK - not exception.

joakim-hove commented on python/res/fm/shell/shell.py:
```diff
@@ -212,3 +212,9 @@ def copyFile(src , target = None):
+    @staticmethod
+    def carefulCopyFile(src , target = None):
+        if os.path.exists(target):
+            raise IOError("Input argument:'%s' already exist" % target)
```
That the target file exists is perfectly OK - not exception.

ManInFez replied:
Should the function fail silently? Would it not be a good idea to let the user know if the copy failed?

joakim-hove replied:
Well - if the file is already present the copy should not proceed, and that is the expected operation of this function - i.e. *not* a failure. Could add a message like:
```
print("File: {} already present - not updated".format(target_file))
```

ManInFez replied:
Very good, I will change it accordingly.

joakim-hove commented on share/ert/forward-models/shell/CAREFUL_COPY_FILE:
```diff
@@ -0,0 +1,9 @@
+STDOUT      careful_copy_file.stdout
+STDERR      careful_copy_file.stderr
```
The std file descriptors are redirected by default - not necessary to do it explicitly.

joakim-hove reviewed (changes requested):
Some minor changes - then she goes in. Good

Commit:
Remove raise exception in carefule_copy_file when file already exists

Printing warning when file exist

Changed unit test of careful_copy_file accordingly
